### ISSUE TITLE
##[error]Error: write EPROTO 4434585024:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:../deps/openssl/openssl/ssl/s23_clnt.c:794:

### ISSUE THREAD

**Issue by ashu16815** [microsoft/google-play-vsts-extension#130]
Hi Team, 
we are facing this issue in Google Play - Increase Rollout whereas Google - Play release and promote is working fine. 

[](url)
[ReleaseLogs_15611.zip](https://github.com/microsoft/google-play-vsts-extension/files/3219142/ReleaseLogs_15611.zip)

Please help here.

**Comment by pjquirk** [comment]
Hi @ashu16815.  That error message indicates an issue in basic SSL negotiation, which can arise from the server requiring a newer version of TLS than the client supports. The "Increase Rollout" task (and only that task) has been updated to use Google's newer APIs, which may require a newer TLS version.  

Are you seeing this on a hosted build agent, or are you using a private agent? Which operating system (and distribution) is it running on?

**Comment by ashu16815** [comment]
WE are using private agent which is running on mac DarwinX64

**Comment by pjquirk** [comment]
My best guess is still that either your agent is outdated and doesn't support the version of TLS that Google requires (probably 1.2). What version of OpenSSL is installed? Check with `openssl version` from the terminal.  It needs to be >=v1.0.1 to support TLS 1.2 according to the [changelog](https://www.openssl.org/news/changelog.html#x35).

Also, are you behind a proxy? See [this issue](https://github.com/googleapis/google-api-nodejs-client/issues/998) for details and possible resolution.

**Comment by xbao** [comment]
Hi @pjquirk, I'm from the same team as @ashu16815.

I see in the discussion of the proxy issue that the fix was merged into googleapis v34.0.0, and the rollout update task depends on v33.0.0 (assuming I'm reading this [package.json](https://github.com/microsoft/google-play-vsts-extension/blob/master/Tasks/google-play-rollout-update/package.json) right)

Would it be possible to get a new version of the task depending on v34.0.0? The [release notes](https://github.com/googleapis/google-api-nodejs-client/releases/tag/v34.0.0) seem pretty innocuous.

**Comment by pjquirk** [comment]
It might be possible, just depends on us being able to get to it.  @lkillgore has more context on the change to using the newere Google APIs.

**Comment by lkillgore** [comment]
Hi @xbao, I'll see about bumping the googleapi version when we come out with our next release.  We haven't started work on it, yet, but we're hoping to get to it very soon.

**Comment by xbao** [comment]
Great, thanks for the updates. 

I don't suppose there's an easy way to have the agent run a local modified version of the task? Then I could do the version bump, maybe it just works or at least I'll find other upgrade related issues early.

**Comment by xbao** [comment]
Hello again.

I see the googleapis dependency was bumped up to v41.0.0 in https://github.com/microsoft/google-play-vsts-extension/commit/31ec451f0973db60122aac95eb248a5bfbe7848c, then reverted to v33.0.0 in https://github.com/microsoft/google-play-vsts-extension/commit/3549702a7d0ee9a915607b258a2bbc807ef05762#diff-dbeb66ad42b202168154cea31140e10b. 

I believe this issue is fixed in googleapis v34.0.0, was there a problem with using the newer version that caused the revert?

**Comment by leantk** [comment]
We will look at making the bump to v34.0.0 then. I'm not sure the details of the failed bump before but I'm marking this to be investigated
